### PR TITLE
[DPE-2205] data-platform-libs v19 external secrets

### PR DESCRIPTION
**Pull request by juditnovak**
## Issue

MySQL K8s wasn't using Relational Secrets that were made available in `data_platform_libs/data_interfaces.py` v18

## Solution

Updateing the libraries, fixing small related issue in the code/tests

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -75,7 +83,29 @@ jobs:
...
         groups: ${{ fromJSON(needs.gh-hosted-collect-integration-tests.outputs.groups) }}
-    name: (GH hosted) ${{ matrix.groups.job_name }}
+        juju-snap-channel:  ["2.9/stable", "3.1/candidate"]
+        include:
+          - juju-snap-channel: "3.1/candidate"
+            agent-version: "3.1.6"
+            libjuju-version:  "3.2.0.1"
+          - juju-snap-channel: "2.9/stable"
+            agent-version: "2.9.44.0"
+            libjuju-version:  "2.9.44.0"
```
Comment (delgod) [inline .github/workflows/ci.yaml:93]:
```suggestion
            agent-version: "2.9.45"
            libjuju-version:  "2.9.44.1"
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -75,7 +83,29 @@ jobs:
       fail-fast: false
       matrix:
         groups: ${{ fromJSON(needs.gh-hosted-collect-integration-tests.outputs.groups) }}
-    name: (GH hosted) ${{ matrix.groups.job_name }}
+        juju-snap-channel:  ["2.9/stable", "3.1/candidate"]
+        include:
+          - juju-snap-channel: "3.1/candidate"
+            agent-version: "3.1.6"
+            libjuju-version:  "3.2.0.1"
```
Comment (delgod) [inline .github/workflows/ci.yaml:90]:
```suggestion
            libjuju-version:  "3.2.2"
```

File: pyproject.toml [hunk 3]
```diff
@@ -57,7 +58,9 @@ pytest-github-secrets = {git = "https://github.com/canonical/data-platform-workf
 pytest-operator = "^0.28.0"
 pytest-operator-cache = {git = "https://github.com/canonical/data-platform-workflows", tag = "v5.0.0", subdirectory = "python/pytest_plugins/pytest_operator_cache"}
 pytest-operator-groups = {git = "https://github.com/canonical/data-platform-workflows", tag = "v5.0.0", subdirectory = "python/pytest_plugins/pytest_operator_groups"}
-juju = "^2.9.44.1"
+juju = "^2.9.44.0 || ^3.2.0.1"
```
Comment (delgod) [inline pyproject.toml:61]:
```suggestion
juju = "^2.9.44.1 || ^3.2.2"
```

File: tests/integration/relations/test_database.py [hunk 4]
```diff
@@ -57,11 +57,11 @@ async def test_build_and_deploy(ops_test: OpsTest):
     # Reduce the update_status frequency until the cluster is deployed
     async with ops_test.fast_forward("60s"):
         await ops_test.model.block_until(
-            lambda: len(ops_test.model.applications[DATABASE_APP_NAME].units) == 3
+            lambda: len(ops_test.model.applications[DATABASE_APP_NAME].units) == 1
```
Comment (delgod) [inline tests/integration/relations/test_database.py:60]: I think we can revert to 3 units in juju 3.1.6

File: tox.ini [hunk 5]
```diff
@@ -74,11 +76,14 @@ pass_env =
...
     GITHUB_OUTPUT
     SECRETS_FROM_GITHUB
+    LIBJUJU_VERSION_SPECIFIER
 allowlist_externals =
     {[testenv:pack-wrapper]allowlist_externals}
 commands_pre =
     poetry install --only integration
     {[testenv:pack-wrapper]commands_pre}
+    # Hacking in after locked deps won't up/downgrade
+    pip install juju=={env:LIBJUJU_VERSION_SPECIFIER:2.9.44.0}
```
Comment (delgod) [inline tox.ini:86]:
```suggestion
    pip install juju=={env:LIBJUJU_VERSION_SPECIFIER:2.9.44.1}
```

File: tests/integration/relations/test_database.py (hunk 4 above)
Comment (juditnovak) [reply tests/integration/relations/test_database.py:60]: My bad, thanks for noticing

File: pyproject.toml (hunk 3 above)
Comment (juditnovak) [reply pyproject.toml]: Original kept instead

**Review by shayancanonical — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**